src/nodes/typeGuards/compound.ts

```diff
@@ -9,15 +9,32 @@ import {
 	isNamespaceBody,
 } from "./union.js";
 
+/**
+ * An `AssertionExpression` that is declared as const.
+ *
+ * @category Node Types
+ */
 export type ConstAssertionExpression = ts.AssertionExpression & {
 	type: ts.TypeReferenceNode;
 	typeName: ConstAssertionIdentifier;
 };
 
+/**
+ * An `Identifier` with an `escapedText` value of `"const"`.
+ *
+ * @category Node Types
+ */
 export type ConstAssertionIdentifier = ts.Identifier & {
 	escapedText: ts.__String & "const";
 };
 
+/**
+ * Test if a node is a {@link ConstAssertionExpression}.
+ *
+ * @category Nodes - Type Guards
+ * @param node - The node in question.
+ * @returns True if the given node appears to be a {@link ConstAssertionExpression}.
```

Comment:
Nit: I've always preferred _"whether"_ instead of _"true/false if"_ for boolean descriptions. Fewer words to work with. Not a blocker IMO.

```suggestion
 * @returns Whether the given node appears to be a {@link ConstAssertionExpression}.
```

_(here and elsewhere)_